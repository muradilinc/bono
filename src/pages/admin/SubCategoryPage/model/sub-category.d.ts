export interface SubCategory {
  id: number;
  name: '';
  parent: number;
}

export interface SubCategoryMutation {
  name: '';
  parent: number | null;
}
