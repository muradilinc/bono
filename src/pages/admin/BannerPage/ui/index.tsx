import { useAppDispatch, useAppSelector } from '../../../../app/store/hooks';
import { selectBanners } from '../../../../features/banner/model/bannerSlice';
import { useEffect } from 'react';
import { getBanners } from '../../../../features/banner/api/bannerThunk';
import { PencilSimple } from '@phosphor-icons/react';
import BannerCard from './BannerCard';
import { Link } from 'react-router-dom';
import { selectGallery } from '../../../../features/gallery/model/gallerySlice';
import { getGallery } from '../../../../features/gallery/api/galleryThunk';

export const AdminBannerPage = () => {
  const banners = useAppSelector(selectBanners);
  const galleries = useAppSelector(selectGallery);
  const dispatch = useAppDispatch();

  useEffect(() => {
    dispatch(getBanners());
    dispatch(getGallery);
  }, [dispatch]);
  console.log(galleries);
  return (
    <section className="bg-black text-white px-[30px] py-[18px] flex flex-col gap-[30px] min-h-[700px]">
      <div className="flex items-center justify-between">
        <h1 className="text-[24px] font-medium">Баннер</h1>
        <Link to="/admin/banner/1">
          <button className="bg-[#2B2B2B] px-[24px] py-[10px] rounded-lg flex items-center">
            <PencilSimple size={24} className="mr-[5px]" />
            Редактировать
          </button>{' '}
        </Link>
      </div>
      {banners.length > 0 ? (
        <div>
          <h1 className="text-[20px] font-medium mt-[10px]">
            ФОТОГРАФИИ ГЛАВНОГО БАННЕРА
          </h1>
          {banners.map((banner) => (
            <BannerCard key={banner.id} banner={banner} />
          ))}
        </div>
      ) : (
        <h1>Пусто</h1>
      )}
      <div>
        <h1 className="text-[20px] font-medium mt-[10px]">
          ФОТОГРАФИИ РЕСТОРАНА
        </h1>
      </div>
    </section>
  );
};
