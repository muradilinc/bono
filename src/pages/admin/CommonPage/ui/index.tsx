import { useEffect, useState } from 'react';
import { useAppDispatch, useAppSelector } from '../../../../app/store/hooks';
import {
  selectSchedulesCommon,
  selectSchedulesCommonLoading,
} from '../../../../features/shedule/model/scheduleSlice';
import {
  deleteBook,
  getSchedulesCommon,
} from '../../../../features/shedule/api/scheduleThunk';
import Loading from '../../../../shared/ui/Loading';
import { Link } from 'react-router-dom';
import { API_LINK } from '../../../../app/constants/links';
import { Trash } from '@phosphor-icons/react';
import { toast } from 'react-toastify';
import ModalDelete from '../../../../shared/ui/ModalDelete';

export const CommonPage = () => {
  const books = useAppSelector(selectSchedulesCommon);
  const loading = useAppSelector(selectSchedulesCommonLoading);
  const dispatch = useAppDispatch();
  const [addModal, setAddModal] = useState<boolean>(false);
  const [id, setId] = useState<number | null>(null);

  const deleteId = (id: number) => {
    setId(id);
    setAddModal(true);
  };
  const onDelete = async () => {
    if (id) {
      await dispatch(deleteBook(id)).unwrap();
      await dispatch(getSchedulesCommon()).unwrap();
      setAddModal(false);
      toast.success('Успешно удалено!');
    } else {
      toast.error('Что то пошло не так!');
    }
  };

  useEffect(() => {
    dispatch(getSchedulesCommon());
  }, [dispatch]);

  if (loading) {
    return <Loading />;
  }

  return (
    <div>
      <div className="w-full h-full bg-black p-8">
        <div className="bg-black p-[16px] flex items-center justify-between">
          <h1 className="font-bold text-[20px] text-white font-comfort">
            Общее
          </h1>
          <Link
            to={API_LINK + '/book/download-booking/'}
            className="bg-[#6BC678] text-white px-[36px] py-[12px] rounded-[4px] capitalize"
          >
            скачать таблицу
          </Link>
        </div>
      </div>
      {books.length > 0 ? (
        <div className="text-white font-medium mt-[30px] overflow-y-scroll max-h-[750px] bookScroll">
          <table className="table-auto w-full text-center">
            <thead className="border-b border-white bg-black sticky top-0">
              <tr>
                <th className="pb-[10px]">№</th>
                <th className="pb-[10px]">Имя</th>
                <th className="pb-[10px]">Номер телефона</th>
                <th className="pb-[10px]">Дата и время</th>
                <th className="pb-[10px]">Длительность</th>
                <th className="pb-[10px]">Кол-во</th>
                <th className="pb-[10px]">Стол</th>
                <th className="pb-[10px]">Комментарий</th>
                <th className="pb-[10px]">Удалить</th>
              </tr>
            </thead>
            <tbody>
              {books.map((book, inx) => (
                <tr key={book.id}>
                  <td className="p-5">{inx + 1}</td>
                  <td>{book.user_name}</td>
                  <td>{book.phone_number}</td>
                  <td className="flex flex-col items-center justify-center">
                    <p>{book.will_come}</p>
                    <p>
                      {book.start_time.slice(0, 5)} -{' '}
                      {book.end_time.slice(0, 5)}
                    </p>
                  </td>
                  <td>{book.time_stamp}</td>
                  <td>{book.amount_guest} пер.</td>
                  <td className="max-w-[100px]">
                    {book.table_set.number_table}
                  </td>
                  <td className="max-w-[100px]">{book.comment}</td>
                  <td className="flex justify-center mt-[10px]">
                    <button
                      onClick={() => deleteId(book.id)}
                      className="flex items-center justify-center text-white bg-[#ff0000ab] rounded-[8px] w-[40px] h-[45px]"
                    >
                      <Trash size={32} />
                    </button>
                  </td>
                </tr>
              ))}
            </tbody>
          </table>
        </div>
      ) : (
        <h4 className="text-white text-center mt-60">Нету данных</h4>
      )}
      {addModal && (
        <ModalDelete
          addModal={addModal}
          setAddModal={setAddModal}
          onDelete={onDelete}
        />
      )}
    </div>
  );
};
