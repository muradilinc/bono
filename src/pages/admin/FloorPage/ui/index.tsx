import { useEffect } from 'react';
import { useAppDispatch, useAppSelector } from '../../../../app/store/hooks';
import {
  selectFloors,
  selectFloorsLoading,
} from '../../../../features/floors/model/floorSlice';
import {
  deleteFloor,
  getFloors,
} from '../../../../features/floors/api/floorThunk';
import { Link } from 'react-router-dom';
import { Pen, Pencil, Trash } from '@phosphor-icons/react';
import { toast } from 'react-toastify';
import Loading from '../../../../shared/ui/Loading';

export const FloorPanel = () => {
  const floors = useAppSelector(selectFloors);
  const loading = useAppSelector(selectFloorsLoading);
  const dispatch = useAppDispatch();

  useEffect(() => {
    dispatch(getFloors());
  }, [dispatch]);

  const handleDeleteFloor = async (id: number) => {
    try {
      await dispatch(deleteFloor(id)).unwrap();
      await dispatch(getFloors()).unwrap();
      toast.success('Отдел успешно удален!');
    } catch (error) {
      console.log(error);
      toast.error('Что-то пошло не так!');
    }
  };
  if (loading) {
    return <Loading />;
  }

  return (
    <>
      <header className="flex items-center justify-between w-full h-[60px] bg-black px-[20px] mt-[10px] flex-wrap">
        <h1 className="text-white font-semibold">Отделы</h1>
        <Link
          to="/admin/department-submit"
          className="font-semibold text-white bg-[#6BC678] rounded-[8px] py-[10px] px-[15px]"
        >
          + Добавить
        </Link>
      </header>

      <section className="bg-black w-full min-h-[635px] py-[30px] px-[20px]">
        <div>
          {floors.length > 0 ? (
            floors.map((floor) => (
              <div
                key={floor.id}
                className="flex items-center justify-between my-[20px] flex-wrap"
              >
                <div className="flex items-center">
                  <div className="text-white ml-[15px]">
                    <div className="flex items-center gap-x-3">
                      <p>Название: {floor.title}</p>
                      <Link to={`/admin/department-submit/${floor.id}`}>
                        <button>
                          <Pen size={22} />
                        </button>
                      </Link>
                    </div>
                  </div>
                </div>
                <div className="flex gap-x-3">
                  <Link
                    className=" text-white bg-red-500 py-[5px] px-[10px] rounded-[5px]"
                    to={`/admin/department-table/${floor.id}`}
                  >
                    <Pencil size={24} />
                  </Link>
                  <button
                    onClick={() => handleDeleteFloor(floor.id)}
                    className=" text-white bg-red-500 py-[5px] px-[10px] rounded-[5px]"
                  >
                    <Trash size={20} />
                  </button>
                </div>
              </div>
            ))
          ) : (
            <p className="text-white">Пусто</p>
          )}
        </div>
      </section>
    </>
  );
};
