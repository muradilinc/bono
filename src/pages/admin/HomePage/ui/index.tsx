import { AdminHeader } from '../../../../features/AdminHeader';

export const AdminHomePage = () => {
  return (
    <>
      <AdminHeader />
    </>
  );
};
