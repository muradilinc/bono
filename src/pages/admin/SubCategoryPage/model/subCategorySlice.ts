import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import {
  getFilterSubcategories,
  getSingleSubCategory,
  getSubCategories,
} from '../api/subCategoryThunk';
import { RootState } from '../../../../app/store/store';
import { SubCategory } from './sub-category';

export interface SubCategoryState {
  subCategories: SubCategory[];
  subCategory: SubCategory | null;
  subCategoriesLoading: boolean;
  subCategoryLoading: boolean;
}

const initialState: SubCategoryState = {
  subCategories: [],
  subCategory: null,
  subCategoriesLoading: false,
  subCategoryLoading: false,
};

const subCategorySlice = createSlice({
  name: 'subCategory',
  initialState,
  reducers: {},
  extraReducers: (builder) => {
    builder.addCase(getSubCategories.pending, (state) => {
      state.subCategoriesLoading = true;
    });
    builder.addCase(
      getSubCategories.fulfilled,
      (state, { payload: subCategories }: PayloadAction<SubCategory[]>) => {
        state.subCategoriesLoading = false;
        state.subCategories = subCategories;
      },
    );
    builder.addCase(getSubCategories.rejected, (state) => {
      state.subCategoriesLoading = false;
    });
    builder.addCase(getSingleSubCategory.pending, (state) => {
      state.subCategoryLoading = true;
    });
    builder.addCase(
      getSingleSubCategory.fulfilled,
      (state, { payload: subCategory }: PayloadAction<SubCategory>) => {
        state.subCategoryLoading = false;
        state.subCategory = subCategory;
      },
    );
    builder.addCase(getSingleSubCategory.rejected, (state) => {
      state.subCategoryLoading = false;
    });
    builder.addCase(getFilterSubcategories.pending, (state) => {
      state.subCategoriesLoading = true;
    });
    builder.addCase(
      getFilterSubcategories.fulfilled,
      (state, { payload: subCategories }: PayloadAction<SubCategory[]>) => {
        state.subCategoriesLoading = false;
        state.subCategories = subCategories;
      },
    );
    builder.addCase(getFilterSubcategories.rejected, (state) => {
      state.subCategoriesLoading = false;
    });
  },
});

export const subCategoryReducer = subCategorySlice.reducer;
export const selectSubCategories = (state: RootState) =>
  state.subCategories.subCategories;
export const selectSubCategoriesLoading = (state: RootState) =>
  state.subCategories.subCategoriesLoading;
export const selectSubCategory = (state: RootState) =>
  state.subCategories.subCategory;
export const selectSubCategoryLoading = (state: RootState) =>
  state.subCategories.subCategoryLoading;
