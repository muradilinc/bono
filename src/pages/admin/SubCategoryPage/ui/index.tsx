import { useEffect } from 'react';
import { useAppDispatch, useAppSelector } from '../../../../app/store/hooks';
import {
  selectSubCategories,
  selectSubCategoriesLoading,
} from '../model/subCategorySlice';
import { deleteSubCategory, getSubCategories } from '../api/subCategoryThunk';
import { Link } from 'react-router-dom';
import { Pen, Trash } from '@phosphor-icons/react';
import { toast } from 'react-toastify';
import Loading from '../../../../shared/ui/Loading';

export const SubCategoriesPage = () => {
  const subCategories = useAppSelector(selectSubCategories);
  const loading = useAppSelector(selectSubCategoriesLoading);
  const dispatch = useAppDispatch();

  useEffect(() => {
    dispatch(getSubCategories());
  }, [dispatch]);

  const handleDeleteCategory = async (id: number) => {
    try {
      await dispatch(deleteSubCategory(id)).unwrap();
      await dispatch(getSubCategories()).unwrap();
      toast.success('Удалено!');
    } catch (error) {
      toast.error('Что-то пошло не так!');
    }
  };

  if (loading) {
    return <Loading />;
  }

  return (
    <>
      <header className="flex items-center justify-between flex-wrap w-full h-[60px] bg-black px-[20px] mt-[10px]">
        <h1 className="text-white font-semibold">Под Категории</h1>
        <Link
          to="/admin/sub-category-submit"
          className="font-semibold text-white bg-[#6BC678] rounded-[8px] py-[10px] px-[15px]"
        >
          + Добавить
        </Link>
      </header>

      <section className="bg-black w-full min-h-[635px] py-[30px] px-[20px]">
        <div>
          {subCategories.length > 0 ? (
            subCategories.map((subCategory) => (
              <div
                key={subCategory.id}
                className="flex items-center justify-between my-[20px] flex-wrap"
              >
                <div className="flex items-center">
                  <div className="text-white ml-[15px]">
                    <div className="flex flex-col gap-x-3">
                      <h4>
                        Категории:{' '}
                        {subCategory.parent_details
                          ? subCategory.parent_details.name
                          : 'Нету!'}
                      </h4>
                      <p>Название: {subCategory.name}</p>
                    </div>
                  </div>
                </div>
                <div className="flex gap-x-3">
                  <Link
                    className="text-white bg-red-500 py-[5px] px-[10px] rounded-[5px]"
                    to={`/admin/sub-category-submit/${subCategory.id}`}
                  >
                    <Pen size={24} />
                  </Link>
                  <button
                    onClick={() => handleDeleteCategory(subCategory.id)}
                    className="text-white bg-red-500 py-[5px] px-[10px] rounded-[5px]"
                  >
                    <Trash size={20} />
                  </button>
                </div>
              </div>
            ))
          ) : (
            <p className="text-white">Пусто</p>
          )}
        </div>
      </section>
    </>
  );
};
