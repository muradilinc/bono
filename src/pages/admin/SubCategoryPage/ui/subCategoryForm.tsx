import { ChangeEvent, FormEvent, useEffect, useState } from 'react';
import { useAppDispatch, useAppSelector } from '../../../../app/store/hooks';
import {
  createSubCategory,
  getSingleSubCategory,
  getSubCategories,
  updateSubCategory,
} from '../api/subCategoryThunk';
import { getCategories } from '../../../../features/category/categoryThunk';
import { selectCategories } from '../../../../features/category/categorySlice';
import { useNavigate, useParams } from 'react-router-dom';
import { SubCategoryMutation } from '../model/sub-category';
import {
  selectSubCategory,
  selectSubCategoryLoading,
} from '../model/subCategorySlice';
import Loading from '../../../../shared/ui/Loading';
import { toast } from 'react-toastify';

export const SubCategoryForm = () => {
  const [subCategory, setSubCategory] = useState<SubCategoryMutation>({
    name: '',
    parent_details: null,
  });
  const categories = useAppSelector(selectCategories);
  const subCategoryApi = useAppSelector(selectSubCategory);
  const loading = useAppSelector(selectSubCategoryLoading);
  const dispatch = useAppDispatch();
  const navigate = useNavigate();
  const { id } = useParams() as { id: string };

  useEffect(() => {
    dispatch(getCategories());
  }, [dispatch]);

  useEffect(() => {
    dispatch(getSingleSubCategory(id));
  }, [dispatch, id]);

  useEffect(() => {
    if (id && subCategoryApi) {
      setSubCategory((prevState) => ({
        ...prevState,
        ...subCategoryApi,
      }));
    }
  }, [id, subCategoryApi]);

  const changeField = (
    event: ChangeEvent<HTMLInputElement | HTMLSelectElement>,
  ) => {
    const { name, value } = event.target;
    setSubCategory((prevState) => ({
      ...prevState,
      [name]: value,
    }));
  };

  const addSubCategoryHandle = async (event: FormEvent) => {
    event.preventDefault();
    try {
      if (id) {
        await dispatch(updateSubCategory({ id, subCategory })).unwrap();
        toast.success('Успешно обновлен!');
      } else {
        await dispatch(createSubCategory(subCategory)).unwrap();
        toast.success('Успешно создан!');
      }
      await dispatch(getSubCategories()).unwrap();
      setSubCategory({
        name: '',
        parent_details: null,
      });
      navigate(-1);
    } catch (error) {
      toast.error('Что-то пошло не так!');
      console.log(error);
    }
  };

  if (loading) {
    return <Loading />;
  }

  return (
    <form onSubmit={addSubCategoryHandle}>
      <header className="flex items-center justify-between w-full h-[60px] bg-black px-[20px]">
        <h1 className="text-white font-semibold">Добавить категорию</h1>
        <button
          type="submit"
          className="font-semibold text-white bg-[#6BC678] rounded-[8px] w-[125px] h-[40px]"
        >
          Сохранить
        </button>
      </header>
      <section className="bg-black w-full min-h-[635px] py-[30px] px-[20px] flex flex-col gap-y-3">
        <div>
          <p className="text-[rgba(255,255,255,0.8)] text-[14px] mb-[5px]">
            Категория
          </p>
          <select
            className="w-full h-[45px] rounded-[8px] px-[10px] text-white bg-[#2B2B2B]"
            name="parent"
            onChange={changeField}
            required
          >
            {subCategory.parent_details ? (
              <option>{subCategory.parent_details.name}</option>
            ) : (
              <option value="">Не выбрано</option>
            )}
            {categories.map((category) => (
              <option id={category.id.toString()} value={category.id}>
                {category.name}
              </option>
            ))}
          </select>
        </div>
        <div>
          <p className="text-[rgba(255,255,255,0.8)] text-[14px] mb-[5px]">
            Название
          </p>
          <input
            value={subCategory.name}
            name="name"
            onChange={changeField}
            required
            className="w-full h-[45px] rounded-[8px] px-[10px] text-white bg-[#2B2B2B]"
            placeholder="Наименование категории"
            type="text"
          />
        </div>
      </section>
    </form>
  );
};
