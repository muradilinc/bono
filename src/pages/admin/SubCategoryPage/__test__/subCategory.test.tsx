import { SubCategoriesPage } from '../ui';
import { renderWithProviders } from '../../../../testUtils/renderWithProviders';
import {
  deleteSubCategory,
  getSingleSubCategory,
  getSubCategories,
} from '../api/subCategoryThunk';
import { configureStore } from '@reduxjs/toolkit';
import { subCategoryReducer } from '../model/subCategorySlice';

describe('SubCategoryPage', () => {
  it('should create sub-category with empty sub-categories', () => {
    const { container } = renderWithProviders(<SubCategoriesPage />);
    expect(container).toMatchSnapshot();
  });
});

describe('getSubcategories', () => {
  describe('getSubcategories', () => {
    global.fetch = jest.fn().mockResolvedValue({
      ok: true,
    });

    it('get sub-categories from  api', async () => {
      const dispatch = jest.fn();
      const thunk = getSubCategories();
      await thunk(dispatch, () => ({}), {});

      const { calls } = dispatch.mock;
      expect(calls).toHaveLength(2);
      const [start, end] = calls;
      expect(start[0].type).toBe(getSubCategories.pending.type);
      expect(end[0].type).toBe(getSubCategories.fulfilled.type);
    });

    it('get single sub-category from api', async () => {
      const store = configureStore({
        reducer: {
          subCategory: subCategoryReducer,
        },
      });
      await store.dispatch(getSubCategories());
      const state = store.getState();
      const firstSubCategory = state.subCategory.subCategories[0].id;

      const dispatch = jest.fn();
      const thunk = getSingleSubCategory(firstSubCategory.toString());
      await thunk(dispatch, () => ({}), {});

      const { calls } = dispatch.mock;
      expect(calls).toHaveLength(2);
      const [start, end] = calls;
      expect(start[0].type).toBe(getSingleSubCategory.pending.type);
      expect(end[0].type).toBe(getSingleSubCategory.fulfilled.type);
    });

    it('delete sub-category', async () => {
      const store = configureStore({
        reducer: {
          subCategory: subCategoryReducer,
        },
      });
      await store.dispatch(getSubCategories());
      const state = store.getState();
      const firstSubCategory = state.subCategory.subCategories[0].id;

      const dispatch = jest.fn();
      const thunk = deleteSubCategory(firstSubCategory);
      await thunk(dispatch, () => ({}), {});

      const { calls } = dispatch.mock;
      expect(calls).toHaveLength(2);

      const [start, end] = calls;
      expect(start[0].type).toBe(deleteSubCategory.pending.type);
      expect(end[0].type).toBe(deleteSubCategory.fulfilled.type);
    });
  });
});
