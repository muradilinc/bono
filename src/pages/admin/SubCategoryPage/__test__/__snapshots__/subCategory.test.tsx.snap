// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubCategoryPage should create sub-category with empty sub-categories 1`] = `
<div>
  <div
    class="fixed flex items-center justify-center w-full h-full bg-black z-[100] flex-col"
  >
    <div
      class="w-12 h-12 rounded-full border-t-3 border-r-3 border-white"
      style="border-top-color: #212121; animation: rotation 1s linear infinite;"
    />
    <span
      class="mt-[10px] text-white text-[14px]"
    >
      Загрузка...
    </span>
  </div>
</div>
`;
